### PR TITLE
Change conditions to leave CL

### PR DESCRIPTION
**Pull request by illusionOfParadise**

### CODE REVIEW

File: src/Controller/EnMarche/CommitteeController.php [hunk 1]
```diff
@@ -108,6 +108,13 @@ public function unfollowAction(Request $request, Committee $committee): Response
             throw $this->createAccessDeniedException('Invalid CSRF protection token to unfollow committee.');
         }
 
+        if ($this->isGranted(CommitteePermissions::SUPERVISE, $committee)
```
Comment (HeahDude) [inline src/Controller/EnMarche/CommitteeController.php]: This controller already check the authorization [here](https://github.com/EnMarche/en-marche.fr/pull/1318/files#diff-7497fcbe29ed5aabeb0342f00e6db8ffR103). Please move this logic in https://github.com/EnMarche/en-marche.fr/blob/master/src/Committee/Voter/FollowCommitteeVoter.php#L47.

File: src/Twig/CommitteeExtension.php [hunk 2]
```diff
@@ -38,6 +38,8 @@ public function getFunctions()
             new \Twig_SimpleFunction('can_create', [$this, 'canCreate']),
             new \Twig_SimpleFunction('can_see', [$this, 'canSee']),
 
+            new \Twig_SimpleFunction('has_several_animators', [$this, 'hasSeveralHostsAndSupervisors']),
```
Comment (HeahDude) [inline src/Twig/CommitteeExtension.php]: No need for this function.

File: templates/committee/show.html.twig [hunk 3]
```diff
@@ -131,7 +131,7 @@
                 <img src="{{ asset('images/loader-sm.gif') }}" id="committee-follow-btn-loader" />
                 Suivre ce comité
             </button>
-        {% elseif can_unfollow(committee) %}
+        {% elseif can_unfollow(committee) and (not is_supervisor(committee) or (is_host and has_several_animators(committee))) %}
```
Comment (HeahDude) [inline templates/committee/show.html.twig]: Changing the voter will allow to keep the check as it is.

File: src/Committee/CommitteeManager.php [hunk 4]
```diff
@@ -127,6 +127,11 @@ public function countCommitteeSupervisors(Committee $committee): int
         return $this->getMembershipRepository()->countSupervisorMembers($committee->getUuid());
     }
 
+    public function hasSeveralHostsAndSupervisors(Committee $committee): int
+    {
+        return $this->getMembershipRepository()->countHostAndSupervisorMembers($committee->getUuid()) > 1;
```
Comment (hhamon) [inline src/Committee/CommitteeManager.php]: I think the `getCommitteeHosts()` method also include supervisors already.

File: src/Controller/EnMarche/CommitteeController.php [hunk 5]
```diff
@@ -108,6 +108,13 @@ public function unfollowAction(Request $request, Committee $committee): Response
...
+            || ($this->isGranted(CommitteePermissions::HOST, $committee)
+                && !$this->get('app.committee.manager')->hasSeveralHostsAndSupervisors($committee))) {
```
Comment (hhamon) [inline src/Controller/EnMarche/CommitteeController.php]: Must be done in the voter to make the logic reusable in controllers and twig templates.

File: src/Controller/EnMarche/CommitteeController.php (hunk 1 above)
Comment (illusionOfParadise) [inline src/Controller/EnMarche/CommitteeController.php]: Done

**Pull request merged**